File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if(nums.length === 0) return 0;

        nums.sort((a,b) => a - b);

        nums = [...new Set(nums)];

        let res = 1;

        let L = 0, R = 1;

        while(R < nums.length){
            if(!(nums[R] - 1 === nums[R - 1])){
                L = R;
            }

            R++;

            res = Math.max(res, R - L);
        }

        return res;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-10.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        l, r = 0, len(nums) - 1

        while l < r:
            
            m = l + (r - l)//2

            mid = nums[m]
            mid_l = nums[m-1]



            if mid < nums[-1]:
                r = m 
            
            else:
                l = m + 1
        
        return nums[l]

    # 4 5 6 7 8 9 0 1 2 3

    # 8 9 0 1 2 3 4 5 6 7 